support-services/src/main/scala/com/gu/support/promotions/dynamo/DynamoService.scala

```diff
@@ -1,41 +1,53 @@
 package com.gu.support.promotions.dynamo
 
 import cats.syntax.either._
-import com.amazonaws.auth.profile.ProfileCredentialsProvider
-import com.amazonaws.auth.{AWSCredentialsProviderChain, InstanceProfileCredentialsProvider}
-import com.amazonaws.regions.Regions
-import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient
-import com.amazonaws.services.dynamodbv2.document.{DynamoDB, Item, Table}
+import com.gu.aws.CredentialsProvider
 import com.typesafe.scalalogging.LazyLogging
 import io.circe.Decoder
 import io.circe.parser._
+import software.amazon.awssdk.enhanced.dynamodb.{
+  AttributeConverterProvider,
+  DynamoDbEnhancedClient,
+  DynamoDbTable,
+  TableSchema,
+}
+import software.amazon.awssdk.enhanced.dynamodb.document.EnhancedDocument
+import software.amazon.awssdk.enhanced.dynamodb.model.Page
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
 
 import scala.jdk.CollectionConverters._
 
-class DynamoService[T](table: Table)(implicit decoder: Decoder[T]) extends LazyLogging {
+class DynamoService[T](table: String)(implicit decoder: Decoder[T]) extends LazyLogging {
```

Comment:
changes in here were a bit tricky, I had to rewrite using the enhanced document api
https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/ddb-en-client-doc-api-steps.html

Reply:
Ah I didn't know about this! I think there was a time where there was no out-of-the-box way to convert to json with v2, hence [code like this](https://github.com/guardian/support-admin-console/blob/main/app/utils/Circe.scala#L34)

Reply:
yes it seems it's heavily requested but only available since last year!